security/subsystem/src/main/java/org/jboss/as/security/service/SimpleSecurityManager.java

```diff
@@ -365,6 +365,26 @@ public void push(final String securityDomain, String userName, char[] password,
         }
     }
 
+    private void propagateSubject(final SecurityContext target, final SecurityContext source) {
+        final SecurityContextUtil previousUtil = source.getUtil();
+        final SecurityContextUtil currentUtil = target.getUtil();
+        if(target.getSecurityDomain() != null && source.getSecurityDomain() != null && target.getSecurityDomain().equals(source.getSecurityDomain())){
```

Comment:
Only a minor comment but shouldn't it be possible to skip the middle && source.getSecurityDomain() or does the final .equals call throw a NPE if null is passed into it?

Reply:
JDoc does not describe what/if of getSecurityDomain. So this is precaution.